staging/src/k8s.io/apiserver/pkg/server/storage/storage_factory.go

```diff
@@ -276,14 +279,42 @@ func (s *DefaultStorageFactory) NewConfig(groupResource schema.GroupResource) (*
 	return storageConfig.ForResource(groupResource), nil
 }
 
-// Backends returns all backends for all registered storage destinations.
-// Used for getting all instances for health validations.
+// Configs implements StorageFactory.
+func (s *DefaultStorageFactory) Configs() []storagebackend.Config {
+	return configs(s.StorageConfig, s.Overrides)
+}
+
+// Returns all storage configurations including those for group resource overrides
+func configs(storageConfig storagebackend.Config, grOverrides map[schema.GroupResource]groupResourceOverrides) (configs []storagebackend.Config) {
+	configs = append(configs, storageConfig)
+	locations := sets.NewString(storageConfig.Transport.ServerList...)
+
```

Comment:
I have some concern about the exact semantics of it.

As an example, with `Backends()` we return all instances from `ServerList`.
With `Configs()` you're passing a single config for that. So effectively we will be pinging a random one from `ServersList`, right?

If so, can you make that explicitly part of specification of the `Config` (in particular both here and in the interfce definition)?

Reply:
Actually, my point was that your original semantic is different than for Backends(), not necessary that this semantic is wrong...

Technically what we really want is that if kube-apiserver can connect to N different etcd instances, we need to ensure that at least one of them is healthy and responsive.

As an example, consider rolling upgrade of etcd instances. There is no point in showing that one of the instances is down, if there are 2 or 4 other that are serving - we can just use those.

So I guess that thinking more about that, the semantic of "config per set of servers" actually make sense to - I just want to ensure that you make it very explicit that it's a different thing from what Backends are doing.

@mborsz - for your thoughts too

Reply:
Agree, just this not intended so I don't want to change it now. I'm even thinking about using `Backends()` as the only difference between `Configs()` and `Backends()` is the function to create a client.